pkg/registry/apis/provisioning/resources/client.go

```diff
@@ -185,65 +201,98 @@ func ForEachResource(ctx context.Context, client dynamic.ResourceInterface, fn f
 	return nil
 }
 
-// ForEachUnmanagedResource applies the function to each unprovisioned supported resource
-func (c *ResourceClients) ForEachUnmanagedResource(ctx context.Context, fn func(client dynamic.ResourceInterface, item *unstructured.Unstructured) error) error {
-	return c.ForEachSupportedResource(ctx, func(client dynamic.ResourceInterface, item *unstructured.Unstructured) error {
-		meta, err := utils.MetaAccessor(item)
-		if err != nil {
-			return fmt.Errorf("extract meta accessor: %w", err)
-		}
+type dummyClients struct{}
+type dummyResourceInterface struct{}
 
-		// Skip if managed
-		_, ok := meta.GetManagerProperties()
-		if ok {
-			return nil
-		}
+func NewDummyClients() ResourceClients {
```

Comment:
@MissingRoberto -- any better suggestions for how to handle this mock client?  We need *something* in test that will resolve a GVK but we do not need a real server/client

Reply:
let's generate with `mockery` (with expecter) automatically now we have have the interfaces :) 


https://github.com/grafana/grafana/blob/773668f5d83222ef79fd9a331dd19cd1039e552a/pkg/services/ngalert/provisioning/persist.go?plain=1#L16-L24